firmware/library/L2_HAL/sensors/temperature/temperature.hpp

```diff
@@ -1,11 +1,74 @@
-// Interface used to sense the temperature of the temperature sensor.
 #pragma once
-#include <cstdint>
+
+#include "L1_Drivers/i2c.hpp"
+#include "utility/bit.hpp"
+
 class TemperatureInterface
 {
  public:
-  virtual bool Initialize() = 0;
-  virtual uint32_t GetTemperatureBytes() = 0;
-  virtual float GetCelsius() = 0;
-  virtual float GetFahrenheit() = 0;
+  virtual bool Initialize()                        = 0;
+  virtual float GetCelsius(I2c i2c_temperature)    = 0;
+  virtual float GetFahrenheit()                    = 0;
 };
+
+class Temperature : public TemperatureInterface
+{
+ public:
+  bool Initialize() override
+  {
+  i2c_.Initialize();
+  uint8_t temperature_sensor_id = 0x00;
+  // Checks to see if the value read by I2c
+  // for the Temperature Chip ID register
+  // is the same as the Base ID (0x14).
+  i2c_.WriteThenRead(kTemperatureAddress, &kTemperatureId,
+  1, &temperature_sensor_id, 1);
+  return (temperature_sensor_id == kMemoryMapAddress);
+  }
+
+  float GetCelsius(I2c i2c_temperature) override
+  {
+  uint8_t one_burst_register         = 0xC4;
+  uint8_t autoinc_bit_register       = 0xC5;
+  uint8_t most_significant_register  = 0xC1;
+  uint8_t least_significant_register = 0xC2;
+  // The register will enable the device to collect data once
+  // and automatically sets the stop bit to 0 (2nd bit).
+  status_ = i2c_temperature.Write(kTemperatureAddress,
+ { one_burst_register, 0x04 }, 2);
+  // Auto increments I2c register address pointer.
+  status_ = i2c_temperature.Write(kTemperatureAddress,
+ { autoinc_bit_register, 0x01 }, 2);
+  i2c_temperature.WriteThenRead(kTemperatureAddress,
+ &most_significant_register, 1, &most_significant_register, 1);
+  i2c_temperature.Read(kTemperatureAddress, &least_significant_register, 1);
+
+  int bit_data = 0;
```

Comment:
```suggestion
  uint16_t bit_data = 0;
```
Please use stdint rather then using int. Please see the style guidelines for SJSU-Dev2 here:
https://sjsu-dev2.readthedocs.io/en/latest/rst/contributing-toc.html

Find the section regarding the proper declaration of integers.